### DISCUSSION TITLE
nuxt proxy for production environment

### DISCUSSION THREAD

**Discussion by binochoi** [nuxt/nuxt#17714] [Questions] [answered]
vite proxy configuration is available only development environment.
in other words, it is not working on build and start command.
am i wrong? or i was missing out anything?

i'm should to set up proxy to nginx because of this problem. it is not bad.
but i want to know any other solution that could to resolve only to nodejs side like middleware proxy and so on.

if you can describe anything, please suggest your method.
thanks in advance !

**Comment by marcusaram** [comment, marked as answer]
I did hit the same issue. We're running a Nuxt 3 project in production with a proxy for the API requests. Remove the Vite proxy (production is not running Vite) and try the next steps.

I did add `http-proxy-middleware` package and added server middleware like this. We use nuxt configuration for the backend url:
nuxt.config.ts
```
    privateRuntimeConfig: {
        BACKEND_URL: process.env.BACKEND_URL || 'http://localhost:8000'
    },
```

server/middlware/api-proxy.js:
```

import { createProxyMiddleware } from 'http-proxy-middleware'
import config from '#config'

const apiProxyMiddleware = createProxyMiddleware('/api/**', {
    target: config.BACKEND_URL,
    changeOrigin: true,
})

export default async (req, res) => {
    // Workaround for h3 not awaiting next.
    await new Promise((resolve, reject) => {
        const next = (err) => {
            if (err) {
                reject(err)
            } else {
                resolve()
            }
        }
        apiProxyMiddleware(req, res, next)
    })
}
```

Hope that it solves your issue.
